internal/releasesources/release_sources.go

```diff
@@ -0,0 +1,510 @@
+					majorMinorVersion, err := common.GetMajorMinorVersion(node.Version)
+					if err != nil {
+						h.log.WithError(err).Warnf("failed to get the major.minor version of: %s", node.Version)
+						continue
+					}
+
+					if openshiftVersion == *majorMinorVersion {
+						// Got a match for OpenShift ReleaseImage
+						newReleaseImage, err := h.createReleaseImage(channel, node.Version, cpuArchitecture)
+						if err != nil {
+							return nil, err
+						}
+						releases = append(releases, newReleaseImage)
+					}
+				}
+			}
+		}
+	}
+
+	h.log.Debugf("Got %d releases", len(releases))
+	return releases, nil
+}
+
+func (h *releaseSourcesHandler) getStaticReleaseImages() []*common.ReleaseImage {
+	dbReleaseImages := []*common.ReleaseImage{}
+	for _, releaseImage := range h.staticReleaseImages {
+		dbReleaseImage := common.ReleaseImage{ReleaseImage: *releaseImage}
+		// For backward compatibility with release images that lack the CPUArchitectures field.
+		if dbReleaseImage.CPUArchitectures == nil {
+			dbReleaseImage.CPUArchitectures = []string{*dbReleaseImage.CPUArchitecture}
+		}
+		dbReleaseImage.DBCPUArchitectures = dbReleaseImage.CPUArchitectures
+		dbReleaseImages = append(dbReleaseImages, &dbReleaseImage)
+	}
+
+	return dbReleaseImages
+}
+
+// mergeDBReleaseImages merges static and dynamic release images together such that
+// static release images have precedence in case of conflicts e.g. which release is default
+func (h *releaseSourcesHandler) mergeDBReleaseImages(staticReleaseImages, dynamicReleaseImages []*common.ReleaseImage) []*common.ReleaseImage {
+	mergedReleaseImages := []*common.ReleaseImage{}
+	staticReleaseImagesReferenceSet := map[string]bool{}
+	defaultStaticReleaseExists := false
+
+	for _, staticRelease := range staticReleaseImages {
+		mergedReleaseImages = append(mergedReleaseImages, staticRelease)
+		staticReleaseImagesReferenceSet[*staticRelease.URL] = true
+		if staticRelease.Default {
+			defaultStaticReleaseExists = true
+		}
+	}
+
+	for _, dynamicRelease := range dynamicReleaseImages {
+		if staticReleaseImagesReferenceSet[*dynamicRelease.URL] {
+			continue
+		}
+		if dynamicRelease.Default && defaultStaticReleaseExists {
+			dynamicRelease.Default = false
+		}
+		mergedReleaseImages = append(mergedReleaseImages, dynamicRelease)
+	}
+
+	return mergedReleaseImages
+}
+
+func (h *releaseSourcesHandler) syncReleaseImagesWithErr() error {
+	err := h.validateReleaseSources()
+	if err != nil {
+		return err
+	}
+
+	err = h.validateStaticReleaseImages()
+	if err != nil {
+		return err
+	}
+	dynamicDBReleaseImages, err := h.getDynamicReleaseImages()
+	if err != nil {
+		return err
+	}
+	h.log.Debugf("Found %d dynamic release Images", len(dynamicDBReleaseImages))
+	supportLevels, err := h.getSupportLevels()
+	if err != nil {
+		return err
+	}
+	err = h.setSupportLevels(dynamicDBReleaseImages, supportLevels)
+	if err != nil {
+		return err
+	}
+	staticDBReleaseImages := h.getStaticReleaseImages()
```

Comment:
static images can change without service restart? maybe we can list them once?

Reply:
AFAIK in `RESTAPI` mode they shouldn't be changed without a restart from the service. Not sure I understand the part about listing them, do you mean logging them for debugging ?